src/Npgsql/NpgsqlDataReader.cs

```diff
@@ -1074,9 +1074,15 @@ public override long GetBytes(int ordinal, long dataOffset, [CanBeNull] byte[] b
         /// Retrieves data as a <see cref="Stream"/>.
         /// </summary>
         /// <param name="ordinal">The zero-based column ordinal.</param>
+        /// <param name="cancellationToken">
+        /// The cancellation instruction, which propagates a notification that operations should be canceled.
+        /// This does not guarantee the cancellation.
+        /// </param>
         /// <returns>The returned object.</returns>
-        public Task<Stream> GetStreamAsync(int ordinal)
+        public Task<Stream> GetStreamAsync(int ordinal, CancellationToken cancellationToken = default)
```

Comment:
It breaks backward compatibility which means that it should be published in the fifth version of Npgsql.

Reply:
That's right, I documented the binary break (although a recompilation would be enough). In my mind we already know we're introducing breaking changes (and so releasing 5 and not 4.1), but maybe we haven't discussed this enough. Are there any specific objections?

By the way, following discussion in https://github.com/dotnet/corefx/issues/35012#issuecomment-484517345 it may be more correct to simply remove `GetStreamAsync()` and `GetTextReaderAsync()`, assuming we implement #2446.

Reply:
I will re-introduce the older overload in the release/4.1 branch after this PR is merged into dev.

Reply:
We can go the way used by ADO.NET (yep, it's a bad way, I know), but it will help to keep compatibility. It's up to you.

Reply:
For APIs that are Npgsql-specific why not do it the better way... Even new ADO.NET APIs are being introduced with default cancellation tokens.